### ISSUE TITLE
Clang silently ignores -mabi with x86-64 target

### ISSUE THREAD

**Issue by Octocontrabass** [llvm/llvm-project#57976]
When using Clang to compile for any x86-64 target, Clang accepts `-mabi=` with any parameter and silently ignores it. [GCC uses the parameter to change the default calling convention.](https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html)

Asking for Clang to support the same options as GCC might be a bit much, but it would be nice to at least get a warning about the unrecognized option.

**Comment by topperc** [comment]
I wonder if this code in `clang/lib/Driver/Driver.cpp` suppressed the unused warning.

```
  A = Args.getLastArg(options::OPT_mabi_EQ);                                     
  if (A && Target.isMIPS()) { 
```

**Comment by topperc** [comment]
Candidate patch https://reviews.llvm.org/D134671

**Comment by topperc** [comment]
This was fixed in d1ad006a8f64bdc17f618deffa9e7c91d82c444d
